sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1121,6 +1125,30 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   }
 
   /**
+   * Create a function name LTRIM for TRIM(Leading), RTRIM for TRIM(Trailing), TRIM for TRIM(BOTH),
+   * otherwise, return the original function identifier.
+   */
+  private def replaceTrimFunction(funcID: FunctionIdentifier, ctx: FunctionCallContext)
+    : FunctionIdentifier = {
+    val opt = ctx.trimOption
+    if (opt != null) {
+      if (ctx.qualifiedName.getText.toLowerCase != "trim") {
+        throw new ParseException(s"The specified function ${ctx.qualifiedName.getText} " +
+          s"doesn't support with option ${opt.getText}.", ctx)
+      }
+      opt.getType match {
+        case SqlBaseParser.BOTH => funcID
+        case SqlBaseParser.LEADING => funcID.copy(funcName = "ltrim")
+        case SqlBaseParser.TRAILING => funcID.copy(funcName = "rtrim")
+        case _ => throw new ParseException(s"Function trim doesn't support with" +
```

Comment:
Nit: remove `s` and add a space at the end ->
`"Function trim doesn't support with "`